doc/architectural/compilation-and-development.md

```diff
@@ -17,7 +17,29 @@ To be documented.
 
 To be documented.
 
-## Documentation
+## Running tests ##
+
+### Regression tests ###
+
+The regression tests are contained in the `regression/` folder.
+They are grouped into directories for each of the tools/modules.
+Each of these contains a directory per test case, containing a C or C++
+file that triggers the bug and a `.desc` file that describes
```

Comment:
This needs updating to encompass Java as well as the fact that there can now be multiple tests (and `.desc` files) in a single directory.

Reply:
Thanks. I hope this documentation will be updated and expanded soon, but it's worth making sure it isn't factually wrong right now.